doc/examples/reconst_qtdmri.py

```diff
@@ -0,0 +1,417 @@
+Finally, gtab contains the qt-dMRI gradient tables for the DWIs in the dataset.
+
+The data consists of 515 DWIs, divided over 35 shells, with 7 "gradient
+strength shells" up to 491 mT/m, 5 equally spaced "pulse separation shells"
+(big_delta) between [10.8-20] ms and a pulse duration (small_delta) of 5ms.
+
+To visualize qt-dMRI acquisition schemes in an intuitive way, the qtdmri module
+provides a visualization function to illustrate the relationship between
+gradient strength (G), pulse separation (big_delta) and b-value:
+"""
+
+plt.figure()
+qtdmri.visualise_gradient_table_G_Delta_rainbow(gtabs[0])
+plt.savefig('qt-dMRI_acquisition_scheme.png')
+
+"""
+.. figure:: qt-dMRI_acquisition_scheme.png
+   :align: center
+
+In the figure the dots represent measured DWIs in any direction, for a given
+gradient strength and pulse separation. The background isolines represent the
+corresponding b-values for different combinations of G and big_delta.
+
+Next, we visualize the middle slices of the test-retest data sets with their
+corresponding masks. To better illustrate the white matter architecture in the
+data, we calculate DTI's fractional anisotropy (FA) over the whole slice and
+project the corpus callosum mask on the FA image.:
+"""
+
+subplot_titles = ["Subject1 Test", "Subject1 Retest",
+                  "Subject2 Test", "Subject2 Tetest"]
+fig = plt.figure()
+plt.subplots(nrows=2, ncols=2)
+for i, (data_, mask_, gtab_) in enumerate(zip(data, cc_masks, gtabs)):
+    # take the middle slice
+    data_middle_slice = data_[:, :, 2]
+    mask_middle_slice = mask_[:, :, 2]
+
+    # estimate fractional anisotropy (FA) for this slice
+    tenmod = dti.TensorModel(gtab_)
+    tenfit = tenmod.fit(data_middle_slice, data_middle_slice[..., 0] > 0)
+    fa = tenfit.fa
+
+    # set mask color to green with 0.5 opacity as overlay
+    mask_template = np.zeros(np.r_[mask_middle_slice.shape, 4])
+    mask_template[mask_middle_slice == 1] = np.r_[0., 1., 0., .5]
+
+    # produce the FA images with corpus callosum masks.
+    plt.subplot(2, 2, 1 + i)
+    plt.title(subplot_titles[i], fontsize=15)
+    plt.imshow(fa, cmap='Greys_r', origin=True, interpolation='nearest')
+    plt.imshow(mask_template, origin=True, interpolation='nearest')
+    plt.axis('off')
+plt.tight_layout()
+plt.savefig('qt-dMRI_datasets_fa_with_ccmasks.png')
+
+"""
+.. figure:: qt-dMRI_datasets_fa_with_ccmasks.png
+   : align: center
+
+Next, we use qt-dMRI to estimate of time-dependent q-space indices
+(q$\tau$-indices) for the masked voxels in the corpus callosum of each dataset.
+In particular, we estimate the Return-to-Original, Return-to-Axis and
+Return-to-Plane Probability (RTOP, RTAP and RTPP), as well as the Mean Squared
+Displacement (MSD).
+
+In this example we don't extrapolate the data beyond the maximum diffusion
+time, so we estimate q$\tau$ indices between the minimum and maximum diffusion
+times of the data at 5 equally spaced points. However, it should the noted that
```

Comment:
I didn't understand this part "However, it should the noted that". Can you please explain it more or correct it?

Reply:
I meant that after fitting qt-dMRI to the data with it's regularization, its coefficients smoothly represent any point in the qtau space. So as in MAPMRI, you can do any interpolation or extrapolation between or beyond the measured data points of the dataset. Extrapolation over q-space is already done to estimate q-space indices (integrals from 0 to infinity over q), but the qt-dMRI basis also allows for extrapolation over tau. I'll put more emphasis on this when I get back home tonight.

Reply:
Thank you for the explanations.

Reply:
No worries. Any other comments?

Reply:
That's perfect. Thank you.